### PR TITLE
Fix/issue 306 and 309 data zoom bug

### LINKED ISSUE #306: [Bug] dataZoom backgroundChart appears outside the background rect
### Version

1.1.0

### Link to Minimal Reproduction

null

### Steps to Reproduce

image

const spec ={
    "type": "scatter",
    "xField": "230726155928016",
    "yField": "230726162719025",
    "data": [
        {
            "id": "data",
            "values": [
                {
                    "30001": "西北",
                    "230726155928016": "2368978",
                    "230726155928024": "西北",
                    "230726160142009": "815039.5979347229",
                    "230726162719025": "53.40000084042549",
                    "230726162719105": "西北"
                },
                {
                    "30001": "中南",
                    "230726155928016": "13157091",
                    "230726155928024": "中南",
                    "230726160142009": "4137415.0951108932",
                    "230726162719025": "232.15000347048044",
                    "230726162719105": "中南"
                },
                {
                    "30001": "东北1",
                    "230726155928016": "8296805",
                    "230726155928024": "东北1",
                    "230726160142009": "2681567.4745378494",
                    "230726162719025": "234.75000359117985",
                    "230726162719105": "东北1"
                },
                {
                    "30001": "西南",
                    "230726155928016": "4503658",
                    "230726155928024": "西南",
                    "230726160142009": "1303124.5089645386",
                    "230726162719025": "134.2000020891428",
                    "230726162719105": "西南"
                },
                {
                    "30001": "华东",
                    "230726155928016": "14550225",
                    "230726155928024": "华东",
                    "230726160142009": "4684506.442247391",
                    "230726162719025": "331.10000520944595",
                    "230726162719105": "华东"
                },
                {
                    "30001": "华北",
                    "230726155928016": "6939587",
                    "230726155928024": "华北",
                    "230726160142009": "2447301.0141382217",
                    "230726162719025": "74.10000109672546",
                    "230726162719105": "华北"
                }
            ],
            "transform": [
                {
                    "type": "fields",
                    "options": {
                        "fields": {
                            "30001": {
                                "alias": "图例项 "
                            },
                            "230726155928016": {
                                "alias": "行 ID"
                            },
                            "230726155928024": {
                                "alias": "地区"
                            },
                            "230726160142009": {
                                "alias": "销售额"
                            },
                            "230726162719025": {
                                "alias": "折扣"
                            },
                            "230726162719105": {
                                "alias": "地区",
                                "domain": [
                                    "西北",
                                    "中南",
                                    "东北1",
                                    "西南",
                                    "华东",
                                    "华北"
                                ]
                            }
                        }
                    }
                }
            ]
        }
    ],
    "dataZoom": [
        {
            "orient": "left",
            width:30,
            padding:{right:10},
            "backgroundColor": {
                "style": {}
            },
            "backgroundChart": {
                "area": {
              visible:false,

                    "style": {
              visible:false,

                    }
                }
            },
            "selectedBackground": {
                "style": {}
            },
            "middleHandler": {
              visible:false,
                "style": {}
            },
            "endHandler": {
                "style": {}
            }
        },
        {
            "orient": "bottom",
            padding:{
top:15
            },
            height:30,
            "backgroundColor": {
                "style": {}
            },
            // "backgroundChart": {
            //     "area": {
            //         "style": {}
            //     }
            // },
            "dragMask": {
                "style": {
                    fill:'red'
                }
            },
            "startHandler": {
                "style": {}
            },
            "endHandler": {
                "style": {}
            }
        }
    ],
    "axes": [
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#8D8D8D"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#f5222d",
                    "lineWidth": 5,
                    "lineDash": [
                        5,
                        5
                    ]
                }
            },
            "orient": "left",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#8D8D8D"
                }
            },
            "title": {
                "visible": false,
                "text": "折扣",
                "style": {
                    "fontSize": 12,
                    "fill": "#363839",
                    "fontWeight": "normal"
                }
            },
            "label": {
                "visible": true,
                "style": {
                    "fontSize": 12,
                    "fill": "#6F6F6F",
                    "angle": 0,
                    "fontWeight": "normal",
                    "textAlign": "center"
                }
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        },
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#3772ff"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#f5222d",
                    "lineWidth": 5,
                    "lineDash": [
                        5,
                        5
                    ]
                }
            },
            "orient": "bottom",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#3772ff"
                }
            },
            "title": {
                "visible": true,
                "text": "行 ID",
                "style": {
                    "fontSize": 16,
                    "fill": "#479044",
                    "fontWeight": "normal"
                }
            },
            "label": {
                "visible": true,
                "style": {
                    "fontSize": 22,
                    "fill": "#6F6F6F",
                    "angle": 10,
                    "fontWeight": "normal",
                    "textAlign": "center"
                }
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        }
    ],
    "label": {
        "visible": true,
        "overlap": {
            "hideOnHit": true,
            "clampForce": true
        },
        "style": {
            "fontSize": 12,
            "fontWeight": "normal",
            "fill": "#363839",
            "stroke": "rgba(255, 255, 255, 0.8)",
            "lineWidth": 2,
            "strokeOpacity": 1
        },
        "offset": 6,
        "position": "top"
    },
    "legends": [
        {
            "type": "discrete",
            "orient": "top",
            "position": "start",
            "layoutType": "normal",
            "visible": true,
            "hover": false,
            "maxRow": 2,
            "title": {
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "item": {
                "spaceRow": 0,
                "spaceCol": 0,
                "padding": {
                    "top": 4,
                    "bottom": 4,
                    "left": 4,
                    "right": 22
                },
                "background": {
                    "style": {
                        "fillOpacity": 0.001
                    }
                },
                "label": {
                    "style": {
                        "fontSize": 12,
                        "fill": "#6F6F6F"
                    }
                },
                "shape": {
                    "style": {
                        "symbolType": "square"
                    }
                }
            },
            "pager": {
                "textStyle": {},
                "handler": {
                    "style": {},
                    "state": {
                        "disable": {}
                    }
                }
            },
            "padding": {
                "top": 0,
                "bottom": 16,
                "left": 0,
                "right": 0
            }
        },
        {
            "type": "size",
            "slidable": true,
            "title": {
                "visible": true,
                "text": "销售额",
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "orient": "top",
            "position": "start",
            "layoutType": "normal",
            "visible": true,
            "field": [
                "230726160142009"
            ],
            "layoutLevel": 50,
            "handlerText": {
                "style": {
                    "maxLineWidth": 80,
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "padding": {
                "top": 0,
                "bottom": 16,
                "left": 0,
                "right": 0
            },
            "rail": {
                "width": 200
            }
        }
    ],
    "seriesField": "230726162719105",
    "color": {
        "field": "230726162719105",
        "type": "ordinal",
        "range": [
            "#2E62F1",
            "#4DC36A",
            "#FF8406",
            "#FFCC00",
            "#4F44CF",
            "#5AC8FA"
        ],
        "specified": {}
    },
    "sizeField": "230726160142009",
    "size": {
        "type": "linear",
        "range": [
            4,
            16
        ]
    },
    "background": "rgba(255, 255, 255, 0)",
    "point": {
        "state": {
            "hover": {
                "cursor": "pointer",
                "fillOpacity": 0.8,
                "stroke": "#58595B",
                "lineWidth": 1,
                "zIndex": 500
            }
        }
    },
    "hover": {
        "enable": true
    },
    "select": {
        "enable": true
    },
    "animation": false,
    "hash": "2bfcf2358b885087a879b3b6e43e4e56"
}

const vchart = new VChart(spec, { dom: CONTAINER_ID });
vchart.renderAsync();

// Just for the convenience of console debugging, DO NOT COPY!
window['vchart'] = vchart;

---------------
image
const spec ={
    "type": "scatter",
    "xField": "230726155928024",
    "yField": "230727110743029",
    "data": [
        {
            "id": "data",
            "values": [
                {
                    "230726155928024": "西北",
                    "230727110743029": "815039.5979347229"
                },
                {
                    "230726155928024": "中南",
                    "230727110743029": "4137415.0951108932"
                },
                {
                    "230726155928024": "东北1",
                    "230727110743029": "2681567.4745378494"
                },
                {
                    "230726155928024": "西南",
                    "230727110743029": "1303124.5089645386"
                },
                {
                    "230726155928024": "华东",
                    "230727110743029": "4684506.442247391"
                },
                {
                    "230726155928024": "华北",
                    "230727110743029": "2447301.0141382217"
                }
            ],
            "transform": [
                {
                    "type": "fields",
                    "options": {
                        "fields": {
                            "230726155928024": {
                                "alias": "地区",
                                "domain": [
                                    "东北1",
                                    "华北",
                                    "华东",
                                    "西北",
                                    "西南",
                                    "中南"
                                ]
                            },
                            "230727110743029": {
                                "alias": "销售额"
                            }
                        }
                    }
                }
            ]
        }
    ],
    "dataZoom": [
        {
            "orient": "bottom",
            "padding": {
                "top": 15
            },
            "height": 28,
            "backgroundColor": {
                "style": {}
            },
            "backgroundChart": {
                "area": {
                    "style": {}
                }
            },
            "selectedBackground": {
                "style": {}
            },
            "startHandler": {
                "style": {}
            },
            "endHandler": {
                "style": {}
            },
            "middleHandler": {
                "visible": false
            }
        },
        {
            "orient": "left",
            "padding": {
                "right": 15
            },
            "width": 28,
            "backgroundColor": {
                "style": {}
            },
            "backgroundChart": {
                "area": {
                    "style": {}
                }
            },
            "selectedBackground": {
                "style": {}
            },
            "startHandler": {
                "style": {}
            },
            "endHandler": {
                "style": {}
            },
            "middleHandler": {
                "visible": false
            }
        }
    ],
    "axes": [
        {
            "type": "band",
            "tick": {
                "visible": false
            },
            "grid": {
                "visible": false,
                "style": {
                    "stroke": "#DADCDD",
                    "lineWidth": 1,
                    "lineDash": [
                        4,
                        2
                    ]
                }
            },
            "orient": "bottom",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#989999"
                }
            },
            "title": {
                "visible": false,
                "text": "地区",
                "style": {
                    "fontSize": 12,
                    "fill": "#363839",
                    "fontWeight": "normal"
                }
            },
            "label": {
                "visible": true,
                "style": {
                    "fontSize": 12,
                    "fill": "#6F6F6F",
                    "angle": 0,
                    "fontWeight": "normal",
                    "textAlign": "center"
                },
                "minGap": 4,
                "flush": true
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "paddingInner": 0.15,
            "paddingOuter": 0.075
        },
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#3772ff"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#DADCDD",
                    "lineWidth": 1,
                    "lineDash": [
                        4,
                        2
                    ]
                }
            },
            "orient": "left",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#3772ff"
                }
            },
            "title": {
                "visible": true,
                "text": "销售额",
                "style": {
                    "fontSize": 16,
                    "fill": "#479044",
                    "fontWeight": "normal"
                }
            },
            "label": {
                "visible": true,
                "style": {
                    "fontSize": 22,
                    "fill": "#6F6F6F",
                    "angle": 0,
                    "fontWeight": "normal",
                    "textAlign": "center"
                }
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        }
    ],
    "label": {
        "visible": true,
        "overlap": {
            "hideOnHit": true,
            "clampForce": true
        },
        "style": {
            "fontSize": 12,
            "fontWeight": "normal",
            "fill": "#363839",
            "stroke": "rgba(255, 255, 255, 0.8)",
            "lineWidth": 2,
            "strokeOpacity": 1
        },
        "offset": 6,
        "position": "top"
    },
    "color": {
        "field": null,
        "type": "ordinal",
        "range": [
            "#2E62F1"
        ]
    },
    "background": "rgba(255, 255, 255, 0)",
    "point": {
        "state": {
            "hover": {
                "cursor": "pointer",
                "fillOpacity": 0.8,
                "stroke": "#58595B",
                "lineWidth": 1,
                "zIndex": 500
            }
        }
    },
    "hover": {
        "enable": true
    },
    "select": {
        "enable": true
    },
    "animation": false,
    "hash": "8e72eaef750d4f2d2cdf71645c7599c7"
}

const vchart = new VChart(spec, { dom: CONTAINER_ID });
vchart.renderAsync();

// Just for the convenience of console debugging, DO NOT COPY!
window['vchart'] = vchart;

### Current Behavior

image
image

### Expected Behavior

Shows inside the background rect.

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

_No response_

### LINKED ISSUE #309: [Bug] dataZoom throw an error in updateSpec.
### Version

1.1.0

### Link to Minimal Reproduction

null

### Steps to Reproduce

const spec ={
    "type": "scatter",
    "xField": "230726155928016",
    "yField": "230726162719025",
    "data": [
        {
            "id": "data",
            "values": [
                {
                    "30001": "西北",
                    "230726155928016": "2368978",
                    "230726155928024": "西北",
                    "230726160142009": "815039.5979347229",
                    "230726162719025": "53.40000084042549",
                    "230726162719105": "西北"
                },
                {
                    "30001": "中南",
                    "230726155928016": "13157091",
                    "230726155928024": "中南",
                    "230726160142009": "4137415.0951108932",
                    "230726162719025": "232.15000347048044",
                    "230726162719105": "中南"
                },
                {
                    "30001": "东北1",
                    "230726155928016": "8296805",
                    "230726155928024": "东北1",
                    "230726160142009": "2681567.4745378494",
                    "230726162719025": "234.75000359117985",
                    "230726162719105": "东北1"
                },
                {
                    "30001": "西南",
                    "230726155928016": "4503658",
                    "230726155928024": "西南",
                    "230726160142009": "1303124.5089645386",
                    "230726162719025": "134.2000020891428",
                    "230726162719105": "西南"
                },
                {
                    "30001": "华东",
                    "230726155928016": "14550225",
                    "230726155928024": "华东",
                    "230726160142009": "4684506.442247391",
                    "230726162719025": "331.10000520944595",
                    "230726162719105": "华东"
                },
                {
                    "30001": "华北",
                    "230726155928016": "6939587",
                    "230726155928024": "华北",
                    "230726160142009": "2447301.0141382217",
                    "230726162719025": "74.10000109672546",
                    "230726162719105": "华北"
                }
            ],
            "transform": [
                {
                    "type": "fields",
                    "options": {
                        "fields": {
                            "30001": {
                                "alias": "图例项 "
                            },
                            "230726155928016": {
                                "alias": "行 ID"
                            },
                            "230726155928024": {
                                "alias": "地区"
                            },
                            "230726160142009": {
                                "alias": "销售额"
                            },
                            "230726162719025": {
                                "alias": "折扣"
                            },
                            "230726162719105": {
                                "alias": "地区",
                                "domain": [
                                    "西北",
                                    "中南",
                                    "东北1",
                                    "西南",
                                    "华东",
                                    "华北"
                                ]
                            }
                        }
                    }
                }
            ]
        }
    ],
    "dataZoom": [
        {
            "orient": "left",
            "backgroundColor": {
                "style": {}
            },
            "backgroundChart": {
                "area": {
                    "style": {}
                }
            },
            "selectedBackground": {
                "style": {}
            },
            "startHandler": {
                "style": {}
            },
            "endHandler": {
                "style": {}
            }
        },
        {
            "orient": "bottom",
            "backgroundColor": {
                "style": {}
            },
            "backgroundChart": {
                "area": {
                    "style": {}
                }
            },
            "selectedBackground": {
                "style": {}
            },
            "startHandler": {
                "style": {}
            },
            "endHandler": {
                "style": {}
            }
        }
    ],
    "axes": [
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#8D8D8D"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#f5222d",
                    "lineWidth": 5,
                    "lineDash": [
                        5,
                        5
                    ]
                }
            },
            "orient": "left",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#8D8D8D"
                }
            },
            "title": {
                "visible": false,
                "text": "折扣",
                "style": {
                    "fontSize": 12,
                    "fill": "#363839",
                    "fontWeight": "normal"
                }
            },
            "label": {
                "visible": true,
                "style": {
                    "fontSize": 12,
                    "fill": "#6F6F6F",
                    "angle": 0,
                    "fontWeight": "normal",
                    "textAlign": "center"
                }
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        },
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#3772ff"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#f5222d",
                    "lineWidth": 5,
                    "lineDash": [
                        5,
                        5
                    ]
                }
            },
            "orient": "bottom",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#3772ff"
                }
            },
            "title": {
                "visible": true,
                "text": "行 ID",
                "style": {
                    "fontSize": 16,
                    "fill": "#479044",
                    "fontWeight": "normal"
                }
            },
            "label": {
                "visible": true,
                "style": {
                    "fontSize": 22,
                    "fill": "#6F6F6F",
                    "angle": 10,
                    "fontWeight": "normal",
                    "textAlign": "center"
                }
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        }
    ],
    "label": {
        "visible": true,
        "overlap": {
            "hideOnHit": true,
            "clampForce": true
        },
        "style": {
            "fontSize": 12,
            "fontWeight": "normal",
            "fill": "#363839",
            "stroke": "rgba(255, 255, 255, 0.8)",
            "lineWidth": 2,
            "strokeOpacity": 1
        },
        "offset": 6,
        "position": "top"
    },
    "legends": [
        {
            "type": "discrete",
            "orient": "top",
            "position": "start",
            "layoutType": "normal",
            "visible": true,
            "hover": false,
            "maxRow": 2,
            "title": {
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "item": {
                "spaceRow": 0,
                "spaceCol": 0,
                "padding": {
                    "top": 4,
                    "bottom": 4,
                    "left": 4,
                    "right": 22
                },
                "background": {
                    "style": {
                        "fillOpacity": 0.001
                    }
                },
                "label": {
                    "style": {
                        "fontSize": 12,
                        "fill": "#6F6F6F"
                    }
                },
                "shape": {
                    "style": {
                        "symbolType": "square"
                    }
                }
            },
            "pager": {
                "textStyle": {},
                "handler": {
                    "style": {},
                    "state": {
                        "disable": {}
                    }
                }
            },
            "padding": {
                "top": 0,
                "bottom": 16,
                "left": 0,
                "right": 0
            }
        },
        {
            "type": "size",
            "slidable": true,
            "title": {
                "visible": true,
                "text": "销售额",
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "orient": "top",
            "position": "start",
            "layoutType": "normal",
            "visible": true,
            "field": [
                "230726160142009"
            ],
            "layoutLevel": 50,
            "handlerText": {
                "style": {
                    "maxLineWidth": 80,
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "padding": {
                "top": 0,
                "bottom": 16,
                "left": 0,
                "right": 0
            },
            "rail": {
                "width": 200
            }
        }
    ],
    "seriesField": "230726162719105",
    "color": {
        "field": "230726162719105",
        "type": "ordinal",
        "range": [
            "#2E62F1",
            "#4DC36A",
            "#FF8406",
            "#FFCC00",
            "#4F44CF",
            "#5AC8FA"
        ],
        "specified": {}
    },
    "sizeField": "230726160142009",
    "size": {
        "type": "linear",
        "range": [
            4,
            16
        ]
    },
    "background": "rgba(255, 255, 255, 0)",
    "point": {
        "state": {
            "hover": {
                "cursor": "pointer",
                "fillOpacity": 0.8,
                "stroke": "#58595B",
                "lineWidth": 1,
                "zIndex": 500
            }
        }
    },
    "hover": {
        "enable": true
    },
    "select": {
        "enable": true
    },
    "animation": false,
    "hash": "2bfcf2358b885087a879b3b6e43e4e56"
}
const spec2={
    "type": "scatter",
    "xField": "230726155928016",
    "yField": "230726162719025",
    "data": [
        {
            "id": "data",
            "values": [
                {
                    "30001": "西北",
                    "230726155928016": "2368978",
                    "230726155928024": "西北",
                    "230726160142009": "815039.5979347229",
                    "230726162719025": "53.40000084042549",
                    "230726162719105": "西北"
                },
                {
                    "30001": "中南",
                    "230726155928016": "13157091",
                    "230726155928024": "中南",
                    "230726160142009": "4137415.0951108932",
                    "230726162719025": "232.15000347048044",
                    "230726162719105": "中南"
                },
                {
                    "30001": "东北1",
                    "230726155928016": "8296805",
                    "230726155928024": "东北1",
                    "230726160142009": "2681567.4745378494",
                    "230726162719025": "234.75000359117985",
                    "230726162719105": "东北1"
                },
                {
                    "30001": "西南",
                    "230726155928016": "4503658",
                    "230726155928024": "西南",
                    "230726160142009": "1303124.5089645386",
                    "230726162719025": "134.2000020891428",
                    "230726162719105": "西南"
                },
                {
                    "30001": "华东",
                    "230726155928016": "14550225",
                    "230726155928024": "华东",
                    "230726160142009": "4684506.442247391",
                    "230726162719025": "331.10000520944595",
                    "230726162719105": "华东"
                },
                {
                    "30001": "华北",
                    "230726155928016": "6939587",
                    "230726155928024": "华北",
                    "230726160142009": "2447301.0141382217",
                    "230726162719025": "74.10000109672546",
                    "230726162719105": "华北"
                }
            ],
            "transform": [
                {
                    "type": "fields",
                    "options": {
                        "fields": {
                            "30001": {
                                "alias": "图例项 "
                            },
                            "230726155928016": {
                                "alias": "行 ID"
                            },
                            "230726155928024": {
                                "alias": "地区"
                            },
                            "230726160142009": {
                                "alias": "销售额"
                            },
                            "230726162719025": {
                                "alias": "折扣"
                            },
                            "230726162719105": {
                                "alias": "地区",
                                "domain": [
                                    "西北",
                                    "中南",
                                    "东北1",
                                    "西南",
                                    "华东",
                                    "华北"
                                ]
                            }
                        }
                    }
                }
            ]
        }
    ],
    "dataZoom": [
        {
            "orient": "bottom",
            "backgroundColor": {
                "style": {}
            },
            "backgroundChart": {
                "area": {
                    "style": {}
                }
            },
            "selectedBackground": {
                "style": {}
            },
            "startHandler": {
                "style": {}
            },
            "endHandler": {
                "style": {}
            }
        }
    ],
    "axes": [
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#8D8D8D"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#f5222d",
                    "lineWidth": 5,
                    "lineDash": [
                        5,
                        5
                    ]
                }
            },
            "orient": "left",
            "visible": false,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#8D8D8D"
                }
            },
            "title": {
                "visible": false,
                "text": "折扣",
                "style": {
                    "fontSize": 12,
                    "fill": "#363839",
                    "fontWeight": "normal"
                }
            },
            "label": {
                "visible": true,
                "style": {
                    "fontSize": 12,
                    "fill": "#6F6F6F",
                    "angle": 0,
                    "fontWeight": "normal",
                    "textAlign": "center"
                }
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        },
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#3772ff"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#f5222d",
                    "lineWidth": 5,
                    "lineDash": [
                        5,
                        5
                    ]
                }
            },
            "orient": "bottom",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#3772ff"
                }
            },
            "title": {
                "visible": true,
                "text": "行 ID",
                "style": {
                    "fontSize": 16,
                    "fill": "#479044",
                    "fontWeight": "normal"
                }
            },
            "label": {
                "visible": true,
                "style": {
                    "fontSize": 22,
                    "fill": "#6F6F6F",
                    "angle": 10,
                    "fontWeight": "normal",
                    "textAlign": "center"
                }
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        }
    ],
    "label": {
        "visible": true,
        "overlap": {
            "hideOnHit": true,
            "clampForce": true
        },
        "style": {
            "fontSize": 12,
            "fontWeight": "normal",
            "fill": "#363839",
            "stroke": "rgba(255, 255, 255, 0.8)",
            "lineWidth": 2,
            "strokeOpacity": 1
        },
        "offset": 6,
        "position": "top"
    },
    "legends": [
        {
            "type": "discrete",
            "orient": "top",
            "position": "start",
            "layoutType": "normal",
            "visible": true,
            "hover": false,
            "maxRow": 2,
            "title": {
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "item": {
                "spaceRow": 0,
                "spaceCol": 0,
                "padding": {
                    "top": 4,
                    "bottom": 4,
                    "left": 4,
                    "right": 22
                },
                "background": {
                    "style": {
                        "fillOpacity": 0.001
                    }
                },
                "label": {
                    "style": {
                        "fontSize": 12,
                        "fill": "#6F6F6F"
                    }
                },
                "shape": {
                    "style": {
                        "symbolType": "square"
                    }
                }
            },
            "pager": {
                "textStyle": {},
                "handler": {
                    "style": {},
                    "state": {
                        "disable": {}
                    }
                }
            },
            "padding": {
                "top": 0,
                "bottom": 16,
                "left": 0,
                "right": 0
            }
        },
        {
            "type": "size",
            "slidable": true,
            "title": {
                "visible": true,
                "text": "销售额",
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "orient": "top",
            "position": "start",
            "layoutType": "normal",
            "visible": true,
            "field": [
                "230726160142009"
            ],
            "layoutLevel": 50,
            "handlerText": {
                "style": {
                    "maxLineWidth": 80,
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "padding": {
                "top": 0,
                "bottom": 16,
                "left": 0,
                "right": 0
            },
            "rail": {
                "width": 200
            }
        }
    ],
    "seriesField": "230726162719105",
    "color": {
        "field": "230726162719105",
        "type": "ordinal",
        "range": [
            "#2E62F1",
            "#4DC36A",
            "#FF8406",
            "#FFCC00",
            "#4F44CF",
            "#5AC8FA"
        ],
        "specified": {}
    },
    "sizeField": "230726160142009",
    "size": {
        "type": "linear",
        "range": [
            4,
            16
        ]
    },
    "background": "rgba(255, 255, 255, 0)",
    "point": {
        "state": {
            "hover": {
                "cursor": "pointer",
                "fillOpacity": 0.8,
                "stroke": "#58595B",
                "lineWidth": 1,
                "zIndex": 500
            }
        }
    },
    "hover": {
        "enable": true
    },
    "select": {
        "enable": true
    },
    "animation": false,
    "hash": "7bf797a052eec40bd3955861c7d8e1f2"
}
const vchart = new VChart(spec, { dom: CONTAINER_ID });
vchart.renderAsync();

setTimeout(()=>{
  vchart.updateSpec(spec2)
},1000)
// Just for the convenience of console debugging, DO NOT COPY!
window['vchart'] = vchart;

### Current Behavior

dataZoom will throw an error when you add or remove a dataZoom in spec, or change the visible of axes and call updateSpec.
image
image

### Expected Behavior

update correctly

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by skie1997**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close #309 
close #306 

### 🐞 Bugserver case id
64cb46de66d8bceee1e24cf1

### 💡 Background and solution
20230802205848_rec_

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix dataZoom backgroundChart data mapping problem |
| 🇨🇳 Chinese | 修复dataZoom背景图表数据映射问题  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

**Comment by kkxxkk2019** [comment]
bugserver case 补充下

File: common/changes/@visactor/vchart/fix-dataZoom-bug_2023-08-02-12-52.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@visactor/vchart",
+      "comment": "fix(dataZoom): fix backgroundChart data mapping problem. fix #306 fix #309",
+      "type": "none"
```
Comment (kkxxkk2019) [inline common/changes/@visactor/vchart/fix-dataZoom-bug_2023-08-02-12-52.json:6]: patch

File: packages/vchart/src/component/data-zoom/data-zoom/data-zoom.ts [hunk 2]
```diff
@@ -68,12 +68,12 @@ export class DataZoom extends DataFilterBaseComponent {
     this._width = this._computeWidth();
     this._height = this._computeHeight();
     // startHandler和endHandler size如果没有配置，则默认跟随background宽 or 高
-    if (this._originalSpec.startHandler?.style?.size ?? true) {
+    if (this._originalSpec?.startHandler?.style?.size ?? true) {
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-zoom/data-zoom.ts:71]: 这个判断，如果用户配置了 size，岂不是不生效了？`isNil(this._originalSpec?.startHandler?.style?.size)` ？

File: packages/vchart/src/component/data-zoom/data-zoom/data-zoom.ts [hunk 3]
```diff
@@ -68,12 +68,12 @@ export class DataZoom extends DataFilterBaseComponent {
...
       this._spec.startHandler.style.size = this._isHorizontal
         ? this._height - this._middleHandlerSize
         : this._width - this._middleHandlerSize;
     }
-    if (this._originalSpec.endHandler?.style?.size ?? true) {
+    if (this._originalSpec?.endHandler?.style?.size ?? true) {
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-zoom/data-zoom.ts:76]: 同上

File: packages/vchart/src/component/data-zoom/data-zoom/data-zoom.ts (hunk 2 above)
Comment (skie1997) [reply packages/vchart/src/component/data-zoom/data-zoom/data-zoom.ts:71]: 是的，这里有问题的

**Review by kkxxkk2019 — APPROVED** [review]

**Pull request merged**